AutomatedLabWorker/AutomatedLabVMWareWorkerNetwork.psm1

```diff
@@ -1,16 +1,21 @@
-﻿#region Get-LWVMWareNetworkSwitch
-function Get-LWVMWareNetworkSwitch
+﻿#region Get-LWVMwareNetworkSwitch
+function Get-LWVMwareNetworkSwitch
 {
     param (
         [Parameter(Mandatory)]
         [AutomatedLab.VirtualNetwork[]]$VirtualNetwork
     )
-	
+
     Write-LogFunctionEntry
 
     foreach ($network in $VirtualNetwork)
     {
-        $network = Get-VDPortgroup -Name $network.Name
+        $network = Get-VirtualPortgroup -Name $network.Name -ErrorAction SilentlyContinue
+
+        if (-not $network)
+        {
+            $network = Get-VDPortgroup -Name $network.Name -ErrorAction SilentlyContinue
```

Comment:
Is there a difference between Get-VirtualPortGroup and Get-VDPortGroup? For consistency, I would suggest using the same cmdlet both times unless they behave differently.

Reply:
You managed to put the finger exactly to the place I'm currently working on.. :-) 

Yes, there is a very big difference between Get-VirtualPortGroup and Get-VDPortGroup in that the first works with a standard vSwitch, the other with a distributed vSwitch. I've already made the type of vSwitch configurable in my local copy (a new VMwareProperties hashtable when creating the network where you can configure "StandardSwitch" or "DistributedSwitch"). I want to ensure that the desired type of switch can be defined and created automatically.

I've struggled with that however as the SwitchType can internally only be defined as internal/external/private; I'm mapping these values onto the VMware switch types and will in the next steps try to implement the network accordingly. 

Raimund already pointed something out to me earlier and he's very right - the main challenge in supporting different hypervisors is that there's a completely different underlying architecture that needs to be worked with. I'm not deeply familar with PowerCLI and am also no VMware expert, so I'm taking my time with this, it will be a while before this code is ready to be checked in.

By now my local code changes are quite exhaustive, but I like to commit changes once a visible improvement has been completed.